src/main/java/com/softech/coursesvc/controller/CourseController.java

```diff
@@ -1,9 +1,11 @@
 package com.softech.coursesvc.controller;
 
+import org.springframework.validation.annotation.Validated;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
+@Validated
```

Comment:
No need to use this way of validations.. add validation logic manually inside method for all endpoints